gwcs/tests/test_wcs.py

```diff
@@ -1443,3 +1403,156 @@ def test_bounding_box_is_returned_F():
+
+    expected_world = [20*u.arcsec, 10*u.nm, 45*u.deg]
+    #expected_world = [15*u.deg, 20*u.nm, 60*u.arcsec]
+    for expected, output in zip(expected_world, output_world):
+        assert_allclose(output, expected.value)
+
+    world_obj = iwcs.pixel_to_world(*input_pixel)
+    assert isinstance(world_obj[0], coord.SkyCoord)
+    assert isinstance(world_obj[1], coord.SpectralCoord)
+
+    assert u.allclose(world_obj[0].spherical.lat, expected_world[0])
+    assert u.allclose(world_obj[0].spherical.lon, expected_world[2])
+    assert u.allclose(world_obj[1], expected_world[1])
+
+    obj_pixel = iwcs.world_to_pixel(*world_obj)
+    assert_allclose(obj_pixel, u.Quantity(input_pixel).to_value(u.pix))
+
+
+def test_reordered_celestial():
+    # This is a spatial model which is ordered lat, lon for the purposes of this test.
+    # Expected lat=45 deg, lon=20 arcsec
+    spatial = models.Multiply(20*u.arcsec/u.pix) & models.Multiply(15*u.deg/u.pix) | models.Mapping((1,0))
+
+    celestial_frame = cf.CelestialFrame(axes_order=(1, 0), unit=(u.arcsec, u.deg),
+                                        reference_frame=coord.ICRS())
+
+    input_frame = cf.CoordinateFrame(2, ["PIXEL"]*2,
+                                     axes_order=list(range(2)), unit=[u.pix]*2)
+
+    iwcs = wcs.WCS(spatial, input_frame, celestial_frame)
+
+    input_pixel = [1*u.pix, 3*u.pix]
+    output_world = iwcs.pixel_to_world_values(*input_pixel)
+    output_pixel = iwcs.world_to_pixel_values(*output_world)
+    assert_allclose(output_pixel, u.Quantity(input_pixel).to_value(u.pix))
+
+    expected_world = [45*u.deg, 20*u.arcsec]
+    assert_allclose(output_world, [e.value for e in expected_world])
+
+    world_obj = iwcs.pixel_to_world(*input_pixel)
+    assert isinstance(world_obj, coord.SkyCoord)
+
+    assert u.allclose(world_obj.spherical.lat, expected_world[0])
+    assert u.allclose(world_obj.spherical.lon, expected_world[1])
+
+    obj_pixel = iwcs.world_to_pixel(world_obj)
+    assert_allclose(obj_pixel, u.Quantity(input_pixel).to_value(u.pix))
+
+
+@pytest.fixture
+def gwcs_with_pipeline_celestial():
+    input_frame = cf.CoordinateFrame(2, ["PIXEL"]*2,
+                                     axes_order=list(range(2)),
+                                     unit=[u.pix]*2,
+                                     name="input")
+
+    spatial = models.Multiply(20*u.arcsec/u.pix) & models.Multiply(15*u.deg/u.pix)
+
+    celestial_frame = cf.CelestialFrame(axes_order=(0, 1), unit=(u.arcsec, u.deg),
+                                        reference_frame=coord.ICRS(), name="celestial")
+
+    custom = models.Shift(1*u.deg) & models.Shift(2*u.deg)
+
+    output_frame = cf.CoordinateFrame(2, ["CUSTOM"]*2,
+                                      axes_order=list(range(2)), unit=[u.arcsec]*2, name="output")
+
+    pipeline = [
+        (input_frame, spatial),
+        (celestial_frame, custom),
+        (output_frame, None),
+    ]
+
+    return wcs.WCS(pipeline)
```

Comment:
Could you move this as a function to the `examples.py` and add the fixture to the `conftest.py`?